### PR TITLE
New package: ncdc-1.20

### PR DESCRIPTION
**Pull request by whoizit**
eiskaltdcpp-qt not support IPv6 hubs 
https://github.com/eiskaltdcpp/eiskaltdcpp/issues/165
https://github.com/eiskaltdcpp/eiskaltdcpp/issues/308
and they say, IPv6 never be in eiskaltdcpp 
https://github.com/eiskaltdcpp/eiskaltdcpp/issues/308#issuecomment-124264074

I've test it right now, it's not support IPv6 hubs on today.
But, in ADC standard they say "The protocol natively supports IPv6." https://en.wikipedia.org/wiki/Advanced_Direct_Connect
So eiskaltdcpp-qt is broken

I realy want to have `ncdc` ncurses client, cause it's works with ADC and IPv6 hubs. Because in our meshnet community in telegram (http://t.me/meshnet) we are using IPv6-only mesh networks (cjdns, yggdrasil)
It's also the same author as `ncdu`

https://dev.yorhel.nl/ncdc
https://g.blicky.net/ncdc.git/

I not really understand difference hostmakedepends and makedepends with depends in https://github.com/void-linux/void-packages/blob/master/Manual.md 
But seems this template works for me.

### CODE REVIEW

File: srcpkgs/ncdc/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'ncdc'
+pkgname=ncdc
+version=1.20
+revision=1
+build_style=configure
+configure_args="--prefix=/usr --with-geoip"
+hostmakedepends="pkg-config glib-devel ncurses-devel sqlite-devel gnutls-devel geoip-devel"
```
Comment (Hoshpak) [inline srcpkgs/ncdc/template]: most if not all of the `-devel` packages schould be in `makedepends`

File: srcpkgs/ncdc/template [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+depends="bzip2"
```
Comment (Hoshpak) [inline srcpkgs/ncdc/template]: why is that needed?

File: srcpkgs/ncdc/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'ncdc'
+pkgname=ncdc
+version=1.20
+revision=1
+build_style=configure
```
Comment (Hoshpak) [inline srcpkgs/ncdc/template]: why not use `gnu-configure`? The package seems to use GNU autotools.

File: srcpkgs/ncdc/template [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+configure_args="--prefix=/usr --with-geoip"
```
Comment (Hoshpak) [inline srcpkgs/ncdc/template]: prefix would be redundant in that case because it is already set by build-style `gnu-configure`.

File: srcpkgs/ncdc/template (hunk 1 above)
Comment (whoizit) [reply srcpkgs/ncdc/template]: I can see as minimum pkg-config and glib-devel in hostmakedepens in many templates

File: srcpkgs/ncdc/template (hunk 2 above)
Comment (whoizit) [reply srcpkgs/ncdc/template]: I don't rly know. I can see this in https://g.blicky.net/ncdc.git/tree/configure.ac
In this time I've add zlib-devel and bzip2-devel to makedepends

File: srcpkgs/ncdc/template (hunk 1 above)
Comment (Hoshpak) [reply srcpkgs/ncdc/template]: Yes. glib-devel contains binaries that need to be executed on the host. If any of these are needed, it's correct to put it in `hostmakedepends`. pkg-config always goes there as it is a binary to be executed on the host. If none of the binaries are needed, it could also be enough to put `libglib-devel` in `makedepends`.

File: srcpkgs/ncdc/template (hunk 3 above)
Comment (whoizit) [reply srcpkgs/ncdc/template]: it's works just as configure, so idk. I set gnu-configure now

File: srcpkgs/ncdc/template (hunk 2 above)
Comment (Hoshpak) [reply srcpkgs/ncdc/template]: Then that should be enough. `depends` would only be needed if the resulting propragm executed the bzip2 binary at runtime. It probably just links the library and uses that.

**Comment by Hoshpak** [comment]
Now cross-compilation fails because it doesn't find glib. Putting `libglib-devel` in `makedepends`. You could also remove `glib-devel` from `hostmakedepends` to check if it is really needed.

**Comment by Hoshpak** [comment]
The `makeheaders` binary is executed during the build and thus needs to be compiled for the host architecture. Adding this function to the template before `post_install` will fix this:

```
pre_build() {
        make CC=$CC_FOR_BUILD CFLAGS=$BUILD_CFLAGS makeheaders
}
```

I was able to cross-compile for aarch64 after this change so I hope travis will confirm everything to be green once that's sorted out.

File: srcpkgs/ncdc/template [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+version=1.20
+revision=1
+build_style=gnu-configure
+configure_args="--with-geoip"
+hostmakedepends="pkg-config"
+makedepends="libglib-devel ncurses-devel sqlite-devel gnutls-devel geoip-devel zlib-devel bzip2-devel"
+short_desc="Modern and lightweight direct connect client with a ncurses interface"
+maintainer="whoami <whoami@systemli.org>"
+license="MIT"
+homepage="http://dev.yorhel.nl/ncdc"
```
Comment (Hoshpak) [inline srcpkgs/ncdc/template]: Redirects to https so I think it's better to also put https here.
